lib/registry/noop_client.go

```diff
@@ -20,41 +20,41 @@ import (
 	"github.com/uber/makisu/lib/docker/image"
 )
 
-// NoopClientFixture implements the registry.Client interface. It returns the empty
+// NoopClient implements the registry.Client interface. It returns the empty
 // distribution manifest on a pull and does nothing on a push.
-type noopClientFixture struct{}
+type NoopClient struct{}
 
-// NoopClientFixture inits a new NoopClientFixture object for testing.
-func NoopClientFixture() Client {
-	return &noopClientFixture{}
+// NoopClient inits a new NoopClient object for testing.
+func NewNoopClient() Client {
+	return &NoopClient{}
 }
 
 // PullImage implements registry.Client.PullImage.
-func (noopClientFixture) Pull(tag string) (*image.DistributionManifest, error) {
+func (NoopClient) Pull(tag string) (*image.DistributionManifest, error) {
```

Comment:
hmm not sure if this is the correct impl of noop client. Normally a function returns nil error you would expect a non-nil value. 

I wonder why it doesnt break the test...

Reply:
Only the `from_step` use a client, and they have their own mocks, they don't use the noop client. The cache manager now uses a LayerClient to reduce the interface surface that it requires.